pinot-spi/src/main/java/org/apache/pinot/spi/utils/builder/TableConfigBuilder.java

```diff
@@ -460,6 +467,7 @@ public TableConfig build() {
     indexingConfig.setAggregateMetrics(_aggregateMetrics);
     indexingConfig.setOptimizeDictionaryForMetrics(_optimizeDictionaryForMetrics);
     indexingConfig.setNoDictionarySizeRatioThreshold(_noDictionarySizeRatioThreshold);
+    indexingConfig.setTierOverwrites(_tierOverwrites);
```

Comment:
why is this only being set in indexingConfig and not in fieldConfigList? or rather - how does one know which overwrites on is providing via the setter?

Reply:
Both tableIndexConfig (i.e. the indexingConfig object here) and fieldConfigList have their own `tierOverwrites`. The fieldConfigList (with tierOverwrites on individual fieldConfig as needed by the caller) is provided to this TableConfigBuilder separately.

Reply:
clarified from offline discussion (method name is a bit confusing as it's on top level TableConfigBuilder, but can only be used to set indexingConfig). This can be addressed separately as it only affects tests.